File: typescript/resistor-color-duo/resistor-color-duo.ts
enum ResistorColors {
    black, brown, red, orange, yellow,
    green, blue, violet, grey, white
}

export const decodedValue = (colors: string[]) =>
    parseInt(colors.slice(0, 2).map(rc => ResistorColors[rc as keyof typeof ResistorColors]).join(""));

File: r/resistor-color-duo/resistor-color-duo.R
colors = c("black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white") # nolint

value = function(resistor_colors) {
  (match(resistor_colors[1], colors) - 1) * 10 + match(resistor_colors[2], colors) - 1
}
